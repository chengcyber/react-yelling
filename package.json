{
  "name": "react-yelling",
  "version": "0.1.0",
  "description": "flexible, configurable, log out every life cycle function",
  "main": "dist/index.js",
  "author": "CHENG LIU <liucheng.tech@outlook.com>",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/kimochg/react-yelling.git"
  },
  "bugs": {
    "url": "https://github.com/kimochg/react-yelling/issues"
  },
  "keywords": [
    "react-component",
    "react utils",
    "rollup",
    "log",
    "life cycle"
  ],
  "scripts": {
    "test": "jest",
    "dev": "rollup -c",
    "build": "cross-env NODE_ENV=production rollup -c"
  },
  "homepage": "https://github.com/kimochg/react-yelling#readme",
  "dependencies": {
    "debug": "^3.1.0"
  },
  "devDependencies": {
    "@babel/core": "^7.0.0-beta.37",
    "@babel/plugin-proposal-object-rest-spread": "^7.0.0-beta.37",
    "@babel/plugin-transform-modules-commonjs": "^7.0.0-beta.37",
    "@babel/plugin-transform-spread": "^7.0.0-beta.37",
    "@babel/preset-env": "^7.0.0-beta.37",
    "@babel/preset-react": "^7.0.0-beta.37",
    "babel-core": "7.0.0-bridge.0",
    "babel-jest": "^22.0.6",
    "cross-env": "^5.1.3",
    "jest": "^22.0.6",
    "react": "^16.2.0",
    "react-test-renderer": "^16.2.0",
    "rollup": "^0.53.4",
    "rollup-plugin-babel": "^4.0.0-beta.0",
    "rollup-plugin-commonjs": "^8.2.6",
    "rollup-plugin-eslint": "^4.0.0",
    "rollup-plugin-node-resolve": "^3.0.2",
    "rollup-plugin-replace": "^2.0.0",
    "rollup-plugin-uglify": "^2.0.1"
  }
}
