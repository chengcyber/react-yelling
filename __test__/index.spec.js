import React from 'react'
import renderer from 'react-test-renderer'
import yell from 'react-yelling'
import { version } from '../package'

/* Link Component For Test */
const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

class Link extends React.Component {
  constructor(props) {
    super(props);

    this._onMouseEnter = this._onMouseEnter.bind(this);
    this._onMouseLeave = this._onMouseLeave.bind(this);

    this.state = {
      class: STATUS.NORMAL,
    };
  }

  _onMouseEnter() {
    this.setState({class: STATUS.HOVERED});
  }

  _onMouseLeave() {
    this.setState({class: STATUS.NORMAL});
  }

  render() {
    return (
      <a
        className={this.state.class}
        href={this.props.page || '#'}
        onMouseEnter={this._onMouseEnter}
        onMouseLeave={this._onMouseLeave}
      >
        {this.props.children}
      </a>
    );
  }
}
Link = yell()(Link)

test('react component works fine', () => {

  const TestComponent = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>,
  )

  let tree = TestComponent.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onMouseEnter();
  // event trigger the rerender
  tree = TestComponent.toJSON();
  expect(tree).toMatchSnapshot();

  TestComponent.update(
    <Link page="https://www.facebook.com">Safe Facebook</Link>,
  )
  // data trigger the rerender
  tree = TestComponent.toJSON();
  expect(tree).toMatchSnapshot();

})

console.log('TEST PACKAGE VERSION: ' + version)

