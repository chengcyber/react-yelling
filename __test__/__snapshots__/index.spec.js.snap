// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react component works fine 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`react component works fine 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`react component works fine 3`] = `
<a
  className="hovered"
  href="https://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Safe Facebook
</a>
`;
